File: SWP.Core/Entities/ServiceTicketDetail.cs
using SWP.Core.GMSAttribute;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SWP.Core.Entities
{
    [GMSTableName("service_ticket_detail")]
    public class ServiceTicketDetail
    {
        [Column("service_ticket_detail_id")]
        [Key]
        public int ServiceTicketDetailId { get; set; }

        [Column("part_id")]
        public int? PartId { get; set; }

        [Column("garage_service_id")]
        public int? GarageServiceId { get; set; }

        [Column("service_ticket_id")]
        public int? ServiceTicketId { get; set; }

        [Column("quantity")]
        public int? Quantity { get; set; }

        [Column("is_deleted")]
        public int IsDeleted { get; set; } = 0;

        // Navigation properties
        public Part? Part { get; set; }
        public GarageService? GarageService { get; set; }
        public ServiceTicket? ServiceTicket { get; set; }
        public ICollection<Warranty> Warranties { get; set; } = new List<Warranty>();
    }
}

File: GarageProject-BE-Springboot-MainFlow/src/main/java/fptu/edu/vn/training/entity/ServiceTicketDetail.java
package fptu.edu.vn.training.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.LinkedHashSet;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "service_ticket_detail")
public class ServiceTicketDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "service_ticket_detail_id")
    private Integer serviceTicketDetailId;

    @Column(name = "part_id")
    private Integer partId;

    @Column(name = "garage_service_id")
    private Integer garageServiceId;

    @Column(name = "service_ticket_id")
    private Integer serviceTicketId;

    @Column(name = "quantity")
    private Integer quantity;

    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Integer isDeleted = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "part_id", insertable = false, updatable = false)
    private Part part;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "garage_service_id", insertable = false, updatable = false)
    private GarageService garageService;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "service_ticket_id", insertable = false, updatable = false)
    private ServiceTicket serviceTicket;

    @Builder.Default
    @OneToMany(mappedBy = "serviceTicketDetail", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = false)
    private Set<Warranty> warranties = new LinkedHashSet<>();
}
